torch/_dynamo/optimizations/training.py

```diff
@@ -277,18 +277,62 @@ def prims_executor(gm, inputs, *, executor):
     return partial(execute, prim_gm, executor=executor)
 
 
+def nvprims_fw_bw_partition_fn(joint_module, joint_inputs):
+    # This function is called once per forward+backward pass of a graph in AOT
+    # Autograd. We use it to set up the nvFuser-specific FX graph that is later
+    # passed to the executor.
+    from functorch.compile import min_cut_rematerialization_partition
+
+    from torch._prims.context import TorchRefsNvfuserCapabilityMode
+    from torch.fx.experimental.proxy_tensor import make_fx
+
+    # AOT Autograd expects arguments of the traced function to be named exactly
+    # "primals, tangents"
+    def func(primals, tangents):
+        return joint_module(primals, tangents)
+
+    # First we trace the graph conditionally decomposing nodes
+    # that can be sent to the nvfuser executor
+    with TorchRefsNvfuserCapabilityMode():
+        prim_gm = make_fx(func)(*joint_inputs)
+
+    # all nvprims for now
+    recomputable_ops = {
```

Comment:
Wondering if we should define this `recomputable_ops` inside `nvfuser_prims.py`, where new nvprims are added? Just to avoid accidentally adding more normalization/reduction ops and having them default in recompute.

Reply:
Or maybe add compulsory markers to nvprims: this function is a reduction, and this one is a normalization, and so on.

Reply:
That sounds like a better idea. So let's put a TODO here and we can clean it up afterwards.